src/coreclr/tools/Common/TypeSystem/Common/MethodDesc.cs

```diff
@@ -27,7 +27,8 @@ public enum EmbeddedSignatureDataKind
     {
         RequiredCustomModifier = 0,
         OptionalCustomModifier = 1,
-        ArrayShape = 2
+        ArrayShape = 2,
+        UnmanagedCallConv = 3,
```

Comment:
Why do we need this as a EmbeddedSignatureDataKind instead of as a MethodSignatureFlags?

I think I might be missing something. It seems like this change applies to the nested arguments of a function pointer, rather than the function pointer itself. Is that right? Does that mean that this is only for function pointers of function pointers?

Reply:
EmbeddedSignatureData is attached to method signatures, but we can decide which method signature it should attach to.

Consider: `void Blah(delegate*unmanaged[Cdecl]<void> x);`. When we're materializing the topmost parent method signature, we set a flag in the parser that it should generate embedded signature data. Embedded signature data is a sidecar that gets attached to the topmost method signature. It can express things in arbitrary type construction nesting. It's really weird. Back to the signature - we want to attach the modopt (and non-meaningful flags) to the topmost EmbeddedSignatureData. If we were to attach it to the nested signature, it would become part of the identity. But we don't want it part of the identity of the `delegate*unmanaged[Cdecl]<void>` type because it was decided that for the type system purposes, the only bit that matters is whether the method is unmanaged (and not the exact calling convention).

Reply:
Ah it was avoiding putting it in the identity that I missed